contract/src/internals/helpers.rs

```diff
@@ -71,4 +71,286 @@ impl DropZone {
+                            false
+                        )
+                    );
+                } else {
+                    // We're not dealing with a promise so we simply execute the function.
+                    self.on_claim_ft(
+                        // Account ID that claimed the linkdrop
+                        account_id, 
+                        // Account ID that funded the linkdrop
+                        drop_data.funder_id, 
+                        // Balance associated with the linkdrop
+                        drop_data.balance, 
+                        // How much storage was freed when the key was claimed
+                        storage_freed,
+                        // FT Data to be used
+                        data,
+                        // Executing the function and treating it NOT like a callback. 
+                        true
+                    );
+                }
+            },
+            DropType::Simple => {
+                promise.unwrap().then(
+                    // Call on_claim_simple with all unspent GAS + min gas for on claim. No attached deposit.
+                    Self::ext(env::current_account_id())
+                    .with_static_gas(MIN_GAS_FOR_ON_CLAIM)
+                    .on_claim_simple(
+                        // Account ID that funded the linkdrop
+                        drop_data.funder_id, 
+                        // Balance associated with the linkdrop
+                        drop_data.balance, 
+                        // How much storage was freed when the key was claimed
+                        storage_freed,
+                    )
+                );
+            }
+        };
+    }
+
+    /// Internal function for transferring funds and executing the callback
+    pub(crate) fn internal_create_account_and_execute(
+        &mut self, 
+        drop_data: Drop, 
+        new_account_id: AccountId, 
+        new_public_key: PublicKey,
+        storage_freed: u128,
+        token_id: Option<String>,
+        storage_for_longest: Option<u128>
+    ) {
+        // CCC to the linkdrop contract to create the account with the desired balance as the linkdrop amount
+        let promise = ext_linkdrop::ext(self.linkdrop_contract.clone())
+            // Attach the balance of the linkdrop along with the exact gas for create account. No unspent GAS is attached.
+            .with_attached_deposit(drop_data.balance.0)
+            .with_static_gas(GAS_FOR_CREATE_ACCOUNT)
+            .with_unused_gas_weight(0)
+            .create_account(
+                new_account_id.clone(),
+                new_public_key,  
+            );
+
+        // Determine what callback we should use depending on the drop type
+        match drop_data.drop_type {
+            DropType::FC(data) => {
+                require!(data.gas_if_straight_execute.is_none(), "cannot call create account if executing FC with specified attached GAS");
+                promise.then(
```

Comment:
Are these promise.then calls the same as the above calls? Can we have 1 method call the other so we only go through all the drop conditions once? It seems like a lot of duplicate conditions and call?